git-branchless-lib/src/core/check_out.rs

```diff
@@ -71,20 +71,36 @@ pub fn check_out_commit(
         render_smartlog,
     } = options;
 
-    let target = match target {
-        None => None,
+    let (target, oid) = match target {
+        None => (None, None),
         Some(CheckoutTarget::Reference(reference_name)) => {
             let categorized_target = CategorizedReferenceName::new(&reference_name);
-            Some(categorized_target.remove_prefix()?)
+            (Some(categorized_target.remove_prefix()?), None)
         }
-        Some(CheckoutTarget::Oid(oid)) => Some(oid.to_string()),
-        Some(CheckoutTarget::Unknown(target)) => Some(target),
+        Some(CheckoutTarget::Oid(oid)) => (Some(oid.to_string()), Some(oid)),
+        Some(CheckoutTarget::Unknown(target)) => (Some(target), None),
     };
 
     if get_undo_create_snapshots(repo)? {
         create_snapshot(effects, git_run_info, repo, event_log_db, event_tx_id)?;
     }
 
+    let target = if let Some(oid) = oid {
+        let all_branches = repo.get_all_local_branches()?;
```

Comment:
To avoid scanning the repo for branches again, we'll want to use the field in `ReferencesSnapshot` called something like `branch_oid_to_names`. Since it's not a parameter to this function, you'll want to refactor this function to pass a `ReferencesSnapshot` from each caller. Then it should also be straightforward to get the matching branches, by just indexing into that `HashMap`.

Also, `ReferencesSnapshot` should be valid for repo states in the past (for `git undo`), unlike the functions on `Repo` which only operate on the present repo state.

Reply:
Looks like I can get `get_branch_oid_to_names` on the existing repo parameter directly by importing `RepoExt`, which I've done here. Let me know if that's the wrong approach